gnomad_qc/v4/analyses/grpmax_comps.py

```diff
@@ -0,0 +1,419 @@
+                    # variants where that group is grpmax
+                    counts_by_thresholds[threshold] = t_ht.aggregate(
+                        hl.agg.counter(t_ht.grpmax_ga)
+                    )
+                    # Add a total count for all groups
+                    counts_by_thresholds[threshold]["all"] = t_ht.count()
+                results_dict[grp_id] = counts_by_thresholds
+        else:
+            logger.info(
+                "Aggregating grpmax variant counts for genetic ancestry groups..."
+            )
+            grps = deepcopy(DIVERSE_GRPS) + ["nfe"]
+            for threshold in AF_THRESHOLDS:
+                t_ht = filter_to_threshold(ht, threshold, version=version)
+                results_dict[threshold] = t_ht.aggregate(
+                    hl.dict(
+                        {grp: hl.agg.count_where(t_ht.grpmax_ga == grp) for grp in grps}
+                    )
+                )
+                results_dict[threshold]["all"] = t_ht.count()
+
+    else:
+        logger.info(
+            "Aggregating variant counts for genetic ancestry groups, variants "
+            "could be counted multiple times if they are above the AF threshold for "
+            "multiple groups..."
+        )
+        grps = deepcopy(DIVERSE_GRPS) + ["nfe"]
+        grp_index_dict = {
+            g: ht.freq_index_dict.get(g + "_adj" if version == "v4" else "gnomad_" + g)
+            for g in grps
+        }
+        for threshold in AF_THRESHOLDS:
+            results_dict[threshold] = ht.aggregate(
+                hl.dict(
+                    {
+                        grp: hl.agg.count_where(ht.freq[idx].AF > threshold)
+                        for grp, idx in grp_index_dict.items()
+                    }
+                )
+            )
+
+    logger.info(f"Results for {version}")
+    pprint(results_dict, sort_dicts=False)
+    return results_dict
+
+
+def create_table(
+    version_dict: Dict[str, Dict[str, Dict[str, int]]],
+    data_subset: str = None,
+    grpmax_counts: bool = False,
+    non_syn_only: bool = False,
+    eur_filter: bool = False,
+):
+    """
+    Create tables of grpmax stats.
+
+    Tables have a column for each version as well as the difference for each grpmax group.
+
+    :param version_dict: Dictionary of grpmax stats
+    :param data_subset: Data subset
+    :param grpmax_counts: Results calculated using only grpmax counts for diverse genetic ancestry groups
+    :param non_syn_only: Keep only non-synonymous variants
+    :param eur_filter: Filter to variants where eur AF is < threshold while grpmax > threshold
+    """
+    headers = ["genetic ancestry group", "v2", "v4", "difference(v4-v2)"]
+    v2_dict_index = (
+        version_dict["v2"][data_subset]
+        if grpmax_counts and eur_filter
+        else version_dict["v2"]
+    )
+    v4_dict_index = (
+        version_dict["v4"][data_subset]
+        if grpmax_counts and eur_filter
+        else version_dict["v4"]
+    )
```

Comment:
not important, but you could do something like
```
versions = ["v2, "v4"]
for version in versions:
    version_dict_idx = (
        version_dict[version][data_subset]
        if grpmax_counts and eur_filter
        else version_dict[version]
    )
```
and then below in the for loop
```
for threshold in AF_THRESHOLDS:
    table = []
    for grp in grps:
        v2_val, v4_val = version_dict_idx[:]
```
though this does remove the `.get`

Reply:
Yeah I didnt love the construction of this piece but youll hit a key error for mid without `get`